src/main/java/org/inferred/freebuilder/processor/excerpt/CheckedBiMap.java

```diff
@@ -48,8 +48,10 @@ public void addTo(SourceBuilder code) {
         .addLine("  }")
         .addLine("")
         .addLine("  @Override public V put(K key, V value) {")
+        .addLine("    K oldKey = biMap.inverse().get(value);")
```

Comment:
get the key, and check if it is null (not set) or the same as the provided key. If not then the value is already used in a bijection.